EIPS/eip-7536.md

```diff
@@ -0,0 +1,226 @@
+ */
+interface IValidator {
+
+    /**
+     * @dev Sets up the validator rules by the edition hash and the data for initialisation. This function will
+     * decode the data back to the required parameters and sets up the rules that decides who can or cannot
+     * mint a copy of the edition.
+     *
+     * @param editionHash The hash of the edition configuration
+     * @param initData The data bytes for initialising the validation rules. Parameters are encoded into bytes
+     */
+    function setRules(
+        bytes32 editionHash, 
+        bytes calldata initData
+    ) external;
+    
+    /**
+     * @dev Supply the data that will be used to validate the fulfilment of the rules setup by the parent token holder.
+     *
+     * @param initiator the party who initiate vadiation
+     * @param editionHash the hash of the edition configuration
+     * @param conditionType the type of condition to validation
+     * @param fullfilmentData the addtion data that is required for passing the validator rules
+     */
+    function validate(
+        address initiator, 
+        bytes32 editionHash,
+        uint256 conditionType,
+        bytes calldata fullfilmentData
+    ) external payable;
+
+}
+```
+
+### Usage
+
+Any ERC-721 token can use the Distributor Interface to create multiple editions, each with a different set of rules to be validated by the Validator Interface before obtaining a child token, and the set of actions that can be invoked after obtaining the child token. This is very useful in community building, and has a diverse range of use cases, for instance:
+
+- Copy Issuance of Unique Artwork/Content: Artists create unique artworks. There could be multiple collectors who want to keep a copy of their artworks. This standard serves as a tool to issue multiple copies of the same kind. The copies can be created with different functions and under different rules. It gives sufficient flexibility to both the Creator and the Collector.
+- Partial Copyright Transfer: This standard enables Creators to conditionally delegate different levels of copyrights, i.e. the right to produce derivative work, to the Collectors. There is no need to sell the original copy, i.e. creator token, in the market. The Creator can instead keep the token as proof of authorship, and the key to managing copy issuance.
+
+People with the following use cases can consider applying this standard:
+
+- Creators of any unique Art/Music NFTs can use this standard to sell copies to audiences. With this standard, they can retain some control over the copies.
+- Artists can use this standard to sell time-limited copies of their artwork to other artists with a copyright statement that enables the production of derivative work
+- Universities can create Graduation Certificates as NFTs and use this standard to mint a batch of non-transferable issues to their students. The Univerity retains the right to revoke any issued certificates, through action.
+
+## Rationale
```

Comment:
Usage belongs more in Rationale than Spec, and the wording can be significantly improved

```suggestion
## Rationale

### Usage

The Distributor Interface enables any ERC-721 token to create multiple editions, each with distinct rules validated by the Validator Interface before minting a child token, and a set of actions available post-minting. This is advantageous for community building, with diverse use cases like:

- **Copy Issuance of Unique Artwork/Content:** Artists can issue multiple copies of their unique artworks with different functions and rules, providing flexibility to both creators and collectors.
- **Partial Copyright Transfer:** Creators can conditionally delegate varying levels of copyrights, allowing for derivative work production without selling the original copy.

Consider using this standard for:

- Selling copies of unique Art/Music NFTs while retaining control over the copies.
- Selling time-limited copies of artwork with attached copyright statements for derivative work production.
- Issuing non-transferable Graduation Certificates as NFTs, where the university retains the revocation right.
```